kartothek/io/dask/dataframe.py

```diff
@@ -301,3 +303,47 @@ def update_dataset_from_ddf(
         metadata=metadata,
         metadata_merger=metadata_merger,
     )
+
+
+def collect_dataset_statistics(
+    store, dataset_uuid, table_name, predicates=None, frac=0.3
+):
+    """
+    Retrieve some statistics, maybe even plot them, generate a report...?
+
+    Parameters
+    ----------
+    store
+      A factory function providing a KeyValueStore
+    dataset_uuid
+      The dataset's unique identifier
+    table_name
+      Name of the kartothek table for which to retrieve the statistics
+    predicates
+      Kartothek predicates to apply filters on the data for which to gather statistics
+    frac
+      Fraction of the data to use for gathering statistics. `frac == 1` will use the whole dataset.
+
+    Returns
+    -------
+    A dask.dataframe containing information about dataset statistics.
+
+    """
+
+    def _get_parquet_stats(mp, store=store, table_name=table_name):
+        """
+        WIP: Retrieve plain parquet metadata, derive some interesting numbers from them
+        """
+        # TODO include more calculations here?
+        pq_metadata = mp.get_parquet_metadata(store=store, table_name=table_name)
+        return pq_metadata
+
+    mp_iterator = dispatch_metapartitions(
```

Comment:
The sampling isn't implemented at the moment, is it? We could probably already push this down into `dispatch_metapartitions`. This may be useful for other applications as well 🤔

Reply:
Yes, apparently I excluded it for testing reasons and forgot to put it back in. Fixed for now.

Reply:
We probably could push that down to the dispatch function, do you have another use case in mind already?

Reply:
> Yes, apparently I excluded it for testing reasons and forgot to put it back in. Fixed for now.

Either way, you should proabably add a test for it ;)

> do you have another use case in mind already?
Sampling actual data would be also a use case. Essentially https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.DataFrame.sample but the sampling would be on IO layer